### PR TITLE
Adds support for configuring PromQL engine evaluation via CLI flags.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Fixes: #426
Fixes: #255

This commits adds support for configuring PromQL engine evaluation.
This widens our PromQL support in order to evaluate queries
that require some specific conditions.

### CODE REVIEW

File: pkg/api/common.go [hunk 1]
```diff
@@ -101,6 +105,12 @@ func ParseFlags(fs *flag.FlagSet, cfg *Config) *Config {
 	fs.StringVar(&cfg.Auth.BasicAuthPasswordFile, "auth-password-file", "", "Path for auth password file containing the actual password used for web endpoint authentication. This flag should be set together with auth-username. It is mutually exclusive with auth-password and bearer-token methods.")
 	fs.StringVar(&cfg.Auth.BearerToken, "bearer-token", "", "Bearer token (JWT) used for web endpoint authentication. Disabled by default. Mutually exclusive with bearer-token-file and basic auth methods.")
 	fs.StringVar(&cfg.Auth.BearerTokenFile, "bearer-token-file", "", "Path of the file containing the bearer token (JWT) used for web endpoint authentication. Disabled by default. Mutually exclusive with bearer-token and basic auth methods.")
+
+	// PromQL configuration flags.
+	fs.DurationVar(&cfg.MaxQueryTimeout, "promql-query-timeout", 2*time.Minute, "Maximum time a query may take before being aborted. Note: The 'timeout' parameter in "+
+		"'/api/v1/query.*' should be within 'promql-query-timeout' range in order to be respected.")
```
Comment (cevian) [inline pkg/api/common.go:111]:
```suggestion
	fs.DurationVar(&cfg.MaxQueryTimeout, "promql-query-timeout", 2*time.Minute, "Maximum time a query may take before being aborted. This option sets both the default and maximum value of the 'timeout' parameter in "+
		"'/api/v1/query.*' endpoints.")
```

File: pkg/api/common.go [hunk 2]
```diff
@@ -101,6 +105,12 @@ func ParseFlags(fs *flag.FlagSet, cfg *Config) *Config {
...
+	fs.DurationVar(&cfg.SubQueryStepInterval, "promql-subquery-step-interval", 1*time.Minute, "Step interval to be used for PromQL subquery evaluation evaluation, "+
+		"if the PromQL subquery does not contain the step value. Example: <metric_name>[30m:]")
```
Comment (cevian) [inline pkg/api/common.go:113]:
```suggestion
	fs.DurationVar(&cfg.SubQueryStepInterval, "promql-subquery-step-interval", 1*time.Minute, "Default step interval to be used for PromQL subquery evaluation. "+
		"This value is used if the subquery does not explicitly specify the step value. Example: <metric_name>[30m:]. Note: in Prometheus this setting is set by the evaluation_interval option.")
```

File: pkg/api/common.go [hunk 3]
```diff
@@ -101,6 +105,12 @@ func ParseFlags(fs *flag.FlagSet, cfg *Config) *Config {
...
 	fs.StringVar(&cfg.Auth.BearerToken, "bearer-token", "", "Bearer token (JWT) used for web endpoint authentication. Disabled by default. Mutually exclusive with bearer-token-file and basic auth methods.")
 	fs.StringVar(&cfg.Auth.BearerTokenFile, "bearer-token-file", "", "Path of the file containing the bearer token (JWT) used for web endpoint authentication. Disabled by default. Mutually exclusive with bearer-token and basic auth methods.")
+
+	// PromQL configuration flags.
+	fs.DurationVar(&cfg.MaxQueryTimeout, "promql-query-timeout", 2*time.Minute, "Maximum time a query may take before being aborted. Note: The 'timeout' parameter in "+
+		"'/api/v1/query.*' should be within 'promql-query-timeout' range in order to be respected.")
+	fs.DurationVar(&cfg.SubQueryStepInterval, "promql-subquery-step-interval", 1*time.Minute, "Step interval to be used for PromQL subquery evaluation evaluation, "+
```
Comment (cevian) [inline pkg/api/common.go:112]: should this be promql-default-subquery-step-interval?

**Review by cevian — APPROVED** [review]
Looks good to me. Just some comments to improve wording.

File: pkg/api/common.go (hunk 3 above)
Comment (Harkishen-Singh) [reply pkg/api/common.go:112]: Yeah, that would be better

**Pull request merged**